### PR TITLE
Added strip() from email/username field. Added e-mail validation whil…

### PR DESCRIPTION
**Pull request by iamgroot42**
…e sharing notes.

### CODE REVIEW

File: app/auth/views.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def before_request():
 def login():
     form = LoginForm()
     if form.validate_on_submit():
-        user = User.query.filter_by(email=form.email.data.lower()).first()
+        user = User.query.filter_by(email=form.email.data.lower().strip()).first()
```
Comment (iamgroot42) [inline app/auth/views.py:28]: strip() to remove trailing and spaces before email/username field.

File: app/auth/views.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def register():
     form = RegistrationForm()
     if form.validate_on_submit():
         user = User(
-            email=form.email.data.lower(),
+            email=form.email.data.lower().strip(),
```
Comment (iamgroot42) [inline app/auth/views.py:49]: strip() to remove trailing and spaces before email/username field.

File: app/auth/views.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def password_reset_request():
         return redirect(url_for('main.index'))
     form = PasswordResetRequestForm()
     if form.validate_on_submit():
-        user = User.query.filter_by(email=form.email.data).first()
+        user = User.query.filter_by(email=form.email.data.lower().strip()).first()
```
Comment (iamgroot42) [inline app/auth/views.py:119]: strip() to remove trailing and spaces before email/username field.

File: app/auth/views.py [hunk 4]
```diff
@@ -135,7 +135,7 @@ def password_reset(token):
         return redirect(url_for('main.index'))
     form = PasswordResetForm()
     if form.validate_on_submit():
-        user = User.query.filter_by(email=form.email.data).first()
+        user = User.query.filter_by(email=form.email.data.lower().strip()).first()
```
Comment (iamgroot42) [inline app/auth/views.py:138]: strip() to remove trailing and spaces before email/username field.

File: app/auth/views.py [hunk 5]
```diff
@@ -152,7 +152,7 @@ def change_email_request():
     form = ChangeEmailForm()
     if form.validate_on_submit():
         if current_user.verify_password(form.password.data):
-            new_email = form.email.data
+            new_email = form.email.data.lower().strip()
```
Comment (iamgroot42) [inline app/auth/views.py:155]: strip() to remove trailing and spaces before email/username field.

File: app/main/forms.py [hunk 6]
```diff
@@ -19,7 +19,7 @@ class NoteForm(Form):
 
 
 class ShareForm(Form):
-    recipient_email = StringField('Recipient Email', validators=[Required()])
+    recipient_email = StringField('Recipient Email', validators=[Required(), Length(1, 254), Email()])
```
Comment (iamgroot42) [inline app/main/forms.py:22]: Email validation while sharing notes via email.

**Comment by levlaz** [comment]
@iamgroot42 you rock! :D

**Comment by iamgroot42** [comment]
@levlaz 

oh__stop_it__you__by_rober_raik-d4cwd9f

**Pull request merged**